test/sqllogictest/transform/fold_constants.slt

```diff
@@ -143,3 +143,650 @@ Explained Query:
+
+query T multiline
+EXPLAIN
+SELECT
+ f1 || f4,
+ UPPER(f1), LOWER(f1),
+ SUBSTRING(f1, 1),
+ REPLACE(f1, f2, f3),
+ POSITION(f2 IN f1),
+ SPLIT_PART(f1, f2, 1),
+ TRANSLATE(f1, 'C', 'Z'),
+ LPAD(f1, 10),
+ LPAD(f1, 10, f2),
+ REGEXP_MATCH(f1, f2),
+ REGEXP_MATCH(f1, f3, 'i') AS case_insensitive,
+ CAST(f1 AS BYTEA),
+ SHA512(CAST(f1 AS BYTEA))
+FROM t_using_dataflow_rendering;
+----
+Explained Query:
+  Project (#6..=#23)
+    Map ((#0 || #3), btrim(#4), ltrim(#4), btrim(#5), rtrim(#5), upper(#0), lower(#0), substr(#0, 1), replace(#0, #1, #2), position(#1, #0), split_string(#0, #1, 1), translate(#0, "C", "Z"), lpad(#0, 10), lpad(#0, 10, #1), regexp_match(#0, #1), regexp_match(#0, #2, "i"), text_to_bytea(#0), digest(#22, "sha512"))
+      Get materialize.public.t_using_dataflow_rendering
+
+EOF
+
+# 4b) using constant folding
+
+statement ok
+CREATE VIEW v_using_constant_folding (f1, f2, f3, f4) AS
+SELECT
+ 'ABCDef', 'CD', 'cd', 'XX';
+
+query TTTTTTTTTTTTTTTTTT
+SELECT
+ f1 || f4,
+ UPPER(f1), LOWER(f1),
+ SUBSTRING(f1, 1),
+ REPLACE(f1, f2, f3),
+ POSITION(f2 IN f1),
+ SPLIT_PART(f1, f2, 1),
+ TRANSLATE(f1, 'C', 'Z'),
+ LPAD(f1, 10),
+ LPAD(f1, 10, f2),
+ REGEXP_MATCH(f1, f2),
+ REGEXP_MATCH(f1, f3, 'i') AS case_insensitive,
+ CAST(f1 AS BYTEA),
+ SHA512(CAST(f1 AS BYTEA))
+FROM v_using_constant_folding;
+----
+ABCDefXX  ABCDef  ABCDef  ABCDef  ABCDef  ABCDEF  abcdef  ABCDef  ABcdef  3  AB  ABZDef      ABCDef  CDCDABCDef  {CD}  {CD}  ABCDef  [238, 226, 119, 230, 210, 45, 158, 105, 232, 127, 11, 125, 76, 10, 244, 149, 96, 158, 231, 207, 230, 136, 216, 56, 30, 157, 228, 74, 89, 49, 203, 21, 238, 196, 151, 235, 205, 247, 59, 41, 81, 87, 67, 152, 170, 149, 41, 38, 39, 103, 214, 97, 156, 131, 169, 93, 142, 69, 43, 29, 72, 113, 255, 236]
+
+query T multiline
+EXPLAIN
+SELECT
+ f1 || f4,
+ UPPER(f1), LOWER(f1),
+ SUBSTRING(f1, 1),
+ REPLACE(f1, f2, f3),
+ POSITION(f2 IN f1),
+ SPLIT_PART(f1, f2, 1),
+ TRANSLATE(f1, 'C', 'Z'),
+ LPAD(f1, 10),
+ LPAD(f1, 10, f2),
+ REGEXP_MATCH(f1, f2),
+ REGEXP_MATCH(f1, f3, 'i') AS case_insensitive,
+ CAST(f1 AS BYTEA),
+ SHA512(CAST(f1 AS BYTEA))
+FROM v_using_constant_folding;
+----
+Explained Query (fast path):
+  Constant
+    - ("ABCDefXX", "ABCDef", "ABCDef", "ABCDef", "ABCDef", "ABCDEF", "abcdef", "ABCDef", "ABcdef", 3, "AB", "ABZDef", "    ABCDef", "CDCDABCDef", {"CD"}, {"CD"}, 0x414243446566, 0xeee277e6d22d9e69e87f0b7d4c0af495609ee7cfe688d8381e9de44a5931cb15eec497ebcdf73b2951574398aa9529262767d6619c83a95d8e452b1d4871ffec)
+
+EOF
+
+# 4c) cleanup
+
+statement ok
+DROP VIEW v_using_constant_folding;
+
+statement ok
+DROP TABLE t_using_dataflow_rendering;
+
+# -------------------------------
+# 5) repeat row (=> this panics!)
```

Comment:
This panics and lets the test crash. How to proceed here?
https://github.com/MaterializeInc/materialize/issues/17758#issuecomment-1438824467 mentions that it requires requires `--unsafe-mode`.